core/arch/arm/sta/interrupt_tests.c

```diff
@@ -113,34 +119,44 @@ struct itr_handler ppi_handler = {
 
 static TEE_Result test_sgi(void)
 {
+	TEE_Result res;
+	uint8_t i, num;
+	uint8_t cpu_mask;
+
 	itr_add(&sgi_handler);
 	itr_enable(TEST_SGI_ID);
 
-	expect_sgi_value[0]++;
-	expect_sgi_value[1]++;
-	expect_sgi_value[2]++;
-	expect_sgi_value[3]++;
-	itr_raise_sgi(TEST_SGI_ID, 0x1 << 0);
-	itr_raise_sgi(TEST_SGI_ID, 0x1 << 1);
-	itr_raise_sgi(TEST_SGI_ID, 0x1 << 2);
-	itr_raise_sgi(TEST_SGI_ID, 0x1 << 3);
+	for (i = 0; i < CFG_TEE_CORE_NB_CORE; i++)
+		expect_sgi_value[i]++;
+	itr_raise_sgi(TEST_SGI_ID, 0xff);
 	tee_time_wait(200);
 	if (memcmp(test_sgi_value, expect_sgi_value, sizeof(test_sgi_value)))
 		return TEE_ERROR_GENERIC;
 
-	expect_sgi_value[0]++;
-	expect_sgi_value[1]++;
-	expect_sgi_value[2]++;
-	expect_sgi_value[3]++;
-	itr_raise_sgi(TEST_SGI_ID, 0xff);
+	res = crypto_ops.prng.read(&num, 1);
```

Comment:
this operation sends sgi to cores #0 up to core #(random).
It seems strange. 
Didn't you expect to send a SGI to a single randomly chosen target core, as implemented in `test_spi()`?

Same question for the test right below in the same routine `test_sgi()`.